### PR TITLE
Port tests to python 3 (0.8)

### PR DESCRIPTION
**Pull request by Arusekk**
This PR targets branch telepathy-haze-0.8

glib2reactor is deprecated in favor of gireactor

### CODE REVIEW

**Comment by Arusekk** [comment]
@Kaffeine ? I don't know the release process, so if you could pull the strings, I hope this can get released soon.

File: tests/twisted/roster/groups.py [hunk 1]
```diff
@@ -154,11 +154,39 @@ def test(q, bus, conn, stream):
...
             )
     assertEquals('duncan@scotland.lit', iq.stanza.query.item['jid'])
-    groups = set([str(x) for x in xpath.queryForNodes('/iq/query/item/group',
-        iq.stanza)])
+    groups = {str(x) for x in xpath.queryForNodes('/iq/query/item/group',
+        iq.stanza)}
     assertLength(1, groups)
     assertContains('Scots', groups)
 
+    # Test SetContactGroups, which didn't previously have proper coverage
```
Comment (Kaffeine) [inline tests/twisted/roster/groups.py:162]: This part seems to leak from the `master` branch.

**Comment by Kaffeine** [comment]
Merged as https://github.com/TelepathyIM/telepathy-haze/commit/6fcaa04e2211f2dda2e4498239360380261b646c

**Comment by Kaffeine** [comment]
Anything else or 0.8.1?

File: tests/twisted/roster/groups.py (hunk 1 above)
Comment (Arusekk) [reply tests/twisted/roster/groups.py:162]: Oops, sorry 😨

**Comment by Arusekk** [comment]
Great! I think this is enough. I now realised that some of this code was actually used at build-time, not only in test phase. Thanks!

**Pull request closed without merging**